lib/kitchen/helpers.rb

```diff
@@ -91,17 +93,45 @@ def data_dockerfile(registry)
     def create_data_image(registry)
       return if ::Docker::Image.exist?(data_image)
 
+      docker_file_content = data_dockerfile(registry)
+      keys_content = insecure_ssh_public_key
+      i = if remote_docker_host?
+            build_docker_image_from_tar(docker_file_content, keys_content)
+          else
+            build_docker_image_from_dir(docker_file_content, keys_content)
+          end
+
+      i.tag("repo" => repo(data_image), "tag" => tag(data_image), "force" => true)
+    end
+
+    def build_docker_image_from_tar(docker_file_content, keys_content)
+      tar_io = StringIO.new
+      Gem::Package::TarWriter.new(tar_io) do |tar|
+        tar.add_file_simple("Dockerfile", 0644, docker_file_content.bytesize) do |io|
+          io.write docker_file_content
+        end
+
+        tar.add_file_simple("authorized_keys", 0644, keys_content.bytesize) do |io|
+          io.write keys_content
+        end
+      end
+
+      tar_io.rewind
+
+      Docker::Image.build_from_tar(tar_io)
```

Comment:
What will be the size of `tar_io`? Everything is handled in memory. As tar_io is a StringIO, it may lead to high memory usage if memory is low in the system

Reply:
The tar will contain only the Dockerfile and the authorized_keys file.  3584 is the length of the tar_io string created here.